### ISSUE TITLE
Automatic Partial Loading

### ISSUE THREAD

**Issue by RodolpheGohard** [angular-translate/angular-translate#771]
# Intro

This post is both a feature request and a call for suggestions about a workaround I designed.

The aim is to have an *Automatic* Partial Loading feature. The partial loading feature of angular-translate look great, but it have the caveat to have you require translations group manually with `$translatePartialLoaderProvider.addPart('home');`.

I have a system which extends angular-translate loading to allow automatic loading of partial translations groups. 

# how it works:

At first, no translations are loaded.

When a translation directive is encountered, it triggers the missing translation handler, since with no translations loaded yet. This is simply how angular translate works. Now our custom handler marks this translation as missing and request the missing translations to be loaded. This is done through a de-bounce function. This allow us to stack successive missing translations, so we can later send a single request for multiple translations.

Once the debounced function starts, it requires all the missing translations we previously stacked. The call is made via `$translate.refresh()` and a custom loader. A trick we use is to require full groups of translations. A backend will select the translations to be sent.

For example, say `home.hello`, `home.welcome`, `home.features` and `menu.title` are missing. Here the custom loader will be called with the parameters `translations?groups=home,menu`. By doing so we don't have to send the full list, plus the whole group is loaded so we also load translations which are likely to be used soon anyway.

We're using this on a starting project, and for now this system works nicely.

Now, I'm writing this because the solution contains several hacks, and sometimes doesn't work well with angular translate features, especially the use of $translate service. I would like some feedback on this feature and the implementation I made.

You can see the implementation here: https://gist.github.com/RodolpheGohard/3a7a4bfc9e3d3ed56988 is not self runnable, but it gives a general idea of how it works.

# TL; DR
angular-translate partial loading is manual, you have to tell it which part you want. I need a system which takes care of which translations groups are needed and which load it for me.

**Comment by knalli** [comment]
Closing due inactivity.

However, I've referenced and taking this idea into account for the 3.0.0 roadmap proposal.
